cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -253,6 +256,12 @@ public KafkaReconciler(
      */
     public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
         return modelWarnings(kafkaStatus)
+                .compose(i ->  {
+                    if (Annotations.booleanAnnotation(kafkaCr, Annotations.ANNO_STRIMZI_IO_BYPASS_BROKER_SCALEDOWN_CHECK, false)) {
```

Comment:
Why don't you extract this in the constructor and store it just asa boolean or do it in the KafkaCluster class instead of storing the whole Kafka CR for the whole reconciliation?